### PR TITLE
Adding the program to calculate square root using Newton Raphson method

### PR DESCRIPTION
**Pull request by Shashwat79802**
***Newton-Raphson Method***

The formula is used to calculate the square root of a number. Added the program in the maths directory and also added the documentation for it in the code.
Let me know if any changes has to be made.

### CODE REVIEW

File: src/main/java/com/thealgorithms/maths/SquareRootWithNewtonRaphsonMethod.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        double x = n;                         //initially taking a guess that x = n.
+        double root;
+
+        while (true) {
+
+            root = 0.5 * (x + n/x);           //applying Newton-Raphson Method.
+
+            if (Math.abs(root - x) < 0.1){    //root - x = error and error < 0.1, 0.1 is the precision value taken over here.
+                break;                        //if error is below the stated precision value, break out of the loop.
+            }
```
Comment (siriak) [inline src/main/java/com/thealgorithms/maths/SquareRootWithNewtonRaphsonMethod.java:41]: Please, move this to loop condition

File: src/main/java/com/thealgorithms/maths/SquareRootWithNewtonRaphsonMethod.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    public static void main(String[] args) {
+
+        Scanner in = new Scanner(System.in);
+        System.out.print("Enter n to find it's square root: ");
+        int n = in.nextInt();
+
+        System.out.println("Square root of " + n + " is " + sqRoot(n));
+    }
+
+    public static double sqRoot (int n) {
```
Comment (siriak) [inline src/main/java/com/thealgorithms/maths/SquareRootWithNewtonRaphsonMethod.java:30]:
```suggestion
    public static double squareRoot (int n) {
```

File: src/main/java/com/thealgorithms/maths/SquareRootWithNewtonRaphsonMethod.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+public class SquareRootWithNewtonRaphsonMethod {
+
+    public static void main(String[] args) {
+
+        Scanner in = new Scanner(System.in);
+        System.out.print("Enter n to find it's square root: ");
+        int n = in.nextInt();
+
+        System.out.println("Square root of " + n + " is " + sqRoot(n));
+    }
```
Comment (siriak) [inline src/main/java/com/thealgorithms/maths/SquareRootWithNewtonRaphsonMethod.java:28]: Please, remove `main()` and add JUnit tests instead

**Review by siriak — CHANGES_REQUESTED** [review]

File: src/main/java/com/thealgorithms/maths/SquareRootWithNewtonRaphsonMethod.java (hunk 3 above)
Comment (Shashwat79802) [reply src/main/java/com/thealgorithms/maths/SquareRootWithNewtonRaphsonMethod.java:28]: Hey Siriak, I'm actually new to it and I don't know about JUnit. I tried to find some helpful tutorials, but they didn't work out for me. What shall I do?

File: src/main/java/com/thealgorithms/maths/SquareRootWithNewtonRaphsonMethod.java (hunk 1 above)
Comment (Shashwat79802) [reply src/main/java/com/thealgorithms/maths/SquareRootWithNewtonRaphsonMethod.java:41]: I didn't get what you're trying to say, but it is in an infinite while loop, and the condition `(Math.abs(root-x) < 0.1)` is the condition for defining the precision of the square root and once it's below 0.1, the loop will break.

Comment (siriak) [reply src/main/java/com/thealgorithms/maths/SquareRootWithNewtonRaphsonMethod.java:41]: `while (Math.abs(root - x) >= 0.1) { root = 0.5 * (x + n/x) }`

File: src/main/java/com/thealgorithms/maths/SquareRootWithNewtonRaphsonMethod.java (hunk 3 above)
Comment (siriak) [reply src/main/java/com/thealgorithms/maths/SquareRootWithNewtonRaphsonMethod.java]: See examples in `src/tests` directory

File: src/main/java/com/thealgorithms/maths/SquareRootWithNewtonRaphsonMethod.java (hunk 1 above)
Comment (Shashwat79802) [reply src/main/java/com/thealgorithms/maths/SquareRootWithNewtonRaphsonMethod.java]: Done.

File: src/main/java/com/thealgorithms/maths/SquareRootWithNewtonRaphsonMethod.java (hunk 3 above)
Comment (Shashwat79802) [reply src/main/java/com/thealgorithms/maths/SquareRootWithNewtonRaphsonMethod.java]: Added JUnit tests.

**Review by siriak — APPROVED** [review]
Looks good, thanks!

**Pull request merged**